Add video conferencing actions to media session article (#5259)
GoogleChrome/web.dev: src/site/content/en/blog/media-session/index.md

To let users know what's currently playing in their browser and control it
without returning to the page that launched it, the Media Session API has been
introduced. It allows web developers to customize this experience through
metadata in custom media notifications, media events such as playing, pausing,

> seeking, track changing, and video conferencing events such as mute/unmute,

Reviewer: nit: "mute/unmute microphone" to distinguish from muting the audio output

Author: Done.

playback from the media hub on desktop so that web developers have a chance to
clear their state.

<b>Example 4:</b> If users are on a video call, they can press the "toggle

> microphone" control in the Picture-in-Picture window to stop website from

Reviewer 2: ```suggestion
microphone" control in the Picture-in-Picture window to stop the website from
```

Author: Done

navigator.mediaSession.setPositionState(null);
```

## Video conferencing actions {: #video-conferencing-actions }

When the user puts their video call into a Picture-in-Picture window, the

> browser may display controls for the microphone, camera, and for hanging up.

Reviewer 2: ```suggestion
browser may display controls for the microphone and camera, and for hanging up.
```

Author: @jpmedley  As you can see in the screenshot below, hanging up is also a control (icon in the PiP window)

![image](https://user-images.githubusercontent.com/634478/117350073-69f1e300-aeac-11eb-9035-7cc7f90d92e4.png)

Author: Done

> browser may display controls for mute/unmute, turnon/turnoff camera, and hang

Reviewer: ditto about mute/unmute microphone

Reviewer 2: This is on the verge of getting wordier than it needs to be to convey the idea (Distinguishing between two types of muting/unmuting will make the whole like clunky). I would go with something like:

>...the browser may display controls the mic, camera, and speaker and for hanging up.

Reviewer 2: I had some typos in that comment, which I've now fixed.

Author: Done.
I've removed `speaker` as it's not supported.

Reviewer 2: LGTM.

When the user clicks those, the website handles them through the video
conferencing actions below.

<figure class="w-figure">
  {% Img
    src="image/vvhSqZboQoZZN9wBvoXq72wzGAf1/fXc7jqc95Oa6sKce7kpZ.jpg",

>     alt="Screenshot of video conferencing controls in a Picture-in-Picture window",

Reviewer 2: ```suggestion
    alt="Screenshot of video conferencing controls in a Picture-in-Picture window.",
```

Author: Done

    width="748",
    height="464"
  %}

>   <figcaption class="w-figcaption">Video conferencing controls in a Picture-in-Picture window</figcaption>

Reviewer 2: ```suggestion
  <figcaption class="w-figcaption">Video conferencing controls in a Picture-in-Picture window.</figcaption>
```

Reviewer 2: I always forget to look for the periods at the ends of figcaption and alt. Please double-check your whole article.

Author: Done for the all article

</figure>

{% Aside %}
At the time of writing, video conferencing actions are available only in
Chrome&nbsp;92 on desktop.
{% endAside %}

### Toggle microphone

> The `"togglemicrophone"` action indicates that the user wants to mute or unmute

Reviewer 2: ```suggestion
The `"togglemicrophone"` action mutes or unmutes
```

Reviewer 2: I thought the wording of these descriptions should be simplified. Is removing 'user' eliminating something useful? If so, I would suggest 'The `"togglemicrophone"` user action' or find some other way to work that in.

Author: The action does not necessarily means the website will do that. That's why it's important to keep "indicates that the user wants to". The website could do anything.

Reviewer 2: > The website could do anything in reply to this action.

Can you make this explicit somehow?

> the microphone. The `setMicrophoneActive(active)` method indicates to the

Reviewer 2: It seems like you should replace `active` with something that indicates this is boolean. I don't really care what you call it.

Author: Would `isActive` work for you?

Author: Done

> the microphone. The `setMicrophoneActive(active)` method indicates to the

Reviewer 2: ```suggestion
the microphone. The `setMicrophoneActive(active)` method tells the
```

Author: Done

> browser whether the microphone is currently considered by the website to be

Reviewer 2: ```suggestion
browser whether the website currently considers the microphone to be
```

Author: Done

active.

```js
let isMicrophoneActive = false;

navigator.mediaSession.setActionHandler('togglemicrophone', () => {
  if (isMicrophoneActive) {
    // Mute the microphone.
  } else {
    // Unmute the microphone.
  }
  isMicrophoneActive = !isMicrophoneActive;
  navigator.mediaSession.setMicrophoneActive(isMicrophoneActive);
});
```

### Toggle camera

> The `"togglecamera"` action indicates that the user wants to turn the active

Reviewer 2: ```suggestion
The `"togglecamera"` action turns the active
```

> camera on or off. The `setCameraActive(active)` method indicates to the browser
> whether the camera is currently considered by the website to be active.

Reviewer 2: ```suggestion
camera on or off. The `setCameraActive(active)` method indicates whether the
browser considers the website to be active.
```

Author: Done

```js
let isCameraActive = false;

navigator.mediaSession.setActionHandler('togglemicrophone', () => {
  if (isCameraActive) {
    // Disable the camera.
  } else {
    // Enable the camera.
  }
  isCameraActive = !isCameraActive;
  navigator.mediaSession.setCameraActive(isCameraActive);
});
```

### Hang up

> The `"hangup"` action indicates that the user wants to end a call.

Reviewer 2: ```suggestion
The `"hangup"` action ends a call.
```
